tomb

```diff
@@ -597,67 +569,60 @@ is_valid_tomb() {
 	}
 	_verbose "tomb file is not empty"
 
-	# no more checking on the uid
-	# _uid="`zstat +uid $1`"
-	# [[ "$_uid"  = "$UID" ]] || {
-	#	  _user="`zstat -s +uid $1`"
-	#	  _warning "Tomb file is owned by another user: ::1 tomb owner::" $_user
-	# }
-	# _verbose "tomb is not owned by another user"
-
-	[[ $_fail = 1 ]] && {
+	[[ $_fail == 1 ]] && {
 		_failure "Tomb command failed: ::1 command name::" $subcommand
 	}
 
-	# TODO: split the rest of that function out.
-	# We already have a valid tomb, now we're checking
-	# whether we can alter it.
-
 	# Tomb file may be a LUKS FS (or we are creating it)
 	[[ "`file $1`" =~ "luks encrypted file" ]] || {
 		_warning "File is not yet a tomb: ::1 tomb file::" $1 }
 
-	_plot $1	 # Set TOMB{PATH,DIR,FILE,NAME}
+	# We set global variables
+	typeset -g TOMBPATH TOMBDIR TOMBFILE TOMBNAME TOMBMAPPER
 
-	# Tomb already mounted (or we cannot alter it)
-	[[ "`_sudo findmnt -rvo SOURCE,TARGET,FSTYPE,OPTIONS,LABEL |
-		awk -vtomb="[$TOMBNAME]" '
-/^\/dev\/mapper\/tomb/ { if($5==tomb) print $1 }'`" = "" ]] || {
-		_failure "Tomb is currently in use: ::1 tomb name::" $TOMBNAME
-	}
-	_verbose "tomb file is not currently in use"
+	TOMBPATH="$1"
 
-	_message "Valid tomb file found: ::1 tomb path::" $TOMBPATH
+	TOMBDIR=$(dirname $TOMBPATH)
 
-	return 0
-}
+	TOMBFILE=$(basename $TOMBPATH)
+
+	# The tomb name is TOMBFILE without an extension and underscores instead of spaces (for mount and cryptsetup)
+	# It can start with dots: ..foo bar baz.tomb -> ..foo_bar_baz
+	TOMBNAME=${${TOMBFILE// /_}%.*}
+	# use the entire filename if the previous transformation returns
+	# an empty string. This handles the corner case of tomb being
+	# hidden files (starting with a dot) and have no extension (only
+	# one dot in string)
+	TOMBNAME=${TOMBNAME:-${TOMBFILE}}
+	[[ "$TOMBNAME" = "" ]] &&
+		_failure "Tomb won't work without a TOMBNAME."
 
-# render the path to the unique /dev/mapper using an hash of the path
-# of the tombfile and its name. Checks for duplicates (tomb is in use)
-render_mapper() {
-	[[ "$tombpath" == "" ]] &&
-		_failure "cannot render mapper: missing \$tombpath"
-	[[ "$TOMBNAME" == "" ]] &&
-		_failure "cannot render mapper: missing \$TOMBNAME"
-	local maphash=`realpath $tombpath | sha256sum -z`
-	mapper="tomb.$TOMBNAME.${maphash[(w)1]}.loop"
+	# checks if Tomb already mounted (or we cannot alter it)
+	local maphash=`realpath $TOMBPATH | sha256sum -z`
+	local nextloop=`losetup -f`
```

Comment:
Right, it seems that if the loop device is not already present in `/dev/` then this line would cause a permission error.

Reply:
oddly enough tests have passed without problems

Reply:
But there the "error" also appears
```
(...)
tomb  .  Testing creation: lock

tomb [D] Identified caller: omnius (1000:100)
tomb [D] Tomb command: lock /tmp/test.tomb
tomb [D] Caller: uid[1000], gid[100], tty[/dev/pts/3].
tomb [D] Temporary directory: /tmp/zsh
tomb [D] is_valid_tomb /tmp/test.tomb
tomb [D] tomb file is readable
tomb [D] tomb file is a regular file
tomb [D] tomb file is not empty
tomb [W] File is not yet a tomb: /tmp/test.tomb
losetup: cannot find an unused loop device: Permission denied
basename: missing operand
Try 'basename --help' for more information.
tomb [D] Mapper: tomb.test.10abda9a4ec9d0e665fe25155685f2e9ddf8f98034aa1b35a8b3748099751b24.
tomb [D] tomb file is not currently in use
tomb  .  Valid tomb file found: /tmp/test.tomb
tomb  .  Commanded to lock tomb test.tomb
tomb [D] Tomb found: /tmp/test.tomb
tomb [D] Loop mounted on /dev/loop0
tomb  .  Checking if the tomb is empty (we never step on somebody else's bones).
tomb  .  Fine, this tomb seems empty.
tomb [D] load_key argument: /tmp/test.tomb.key
tomb [D] load_key: /tmp/test.tomb.key
tomb [D] is_valid_key
(...)
```
In the end it passed the test, but didn't work correctly